test/e2e/capi_test.go

```diff
@@ -275,7 +276,7 @@ var _ = Describe("Running the Cluster API E2E tests", func() {
 				BootstrapClusterProxy:    bootstrapClusterProxy,
 				ArtifactFolder:           artifactFolder,
 				ControlPlaneMachineCount: pointer.Int64(3),
-				WorkerMachineCount:       pointer.Int64(1),
+				WorkerMachineCount:       pointer.Int64(0),
```

Comment:
I should probably remember this but why is this going to zero?

Reply:
CAPI 1.1.0 deprecated the KCP only upgrade tests in favor of using the cluster upgrade tests. However when we bumped CAPI to 1.1.0-beta.1 there was a [bug](https://github.com/kubernetes-sigs/cluster-api/issues/4896#issuecomment-1010523122) which prevented us from using worker machine count 0 so I set to 1 temporarily. Now that the bug is [fixed](https://github.com/kubernetes-sigs/cluster-api/pull/5926), setting it back to 0 to optimize test duration without reducing coverage (we have a separate full cluster upgrade spec above that tests control plane + worker node upgrades, while these two specs are specific to control plane upgrade only with different rolling strategies).

Reply:
Since we are waiting for the release should we make comment in here that points to this explanation or gives one? I know I will forget this 6 months from now.

Reply:
we're not waiting for the release anymore, this was a temporary workaround when we upgraded to beta.1, now that it's fixed we're setting the count to 0 as desired